### PR TITLE
Extend documentation for integration tests

### PR DESCRIPTION
**Pull request by rolandkakonyi**
## Description

There is not much documentation about player testing in the repo right now.

## Changes

- Updated `integration_test/README.md`
- Updated `CONTRIBUTING.md` with instructions for adding new tests
- Added API documentation for player testing functions.

## Checklist
- [x] 🗒 `CHANGELOG` entry

### CODE REVIEW

File: integration_test/playertesting/PlayerTesting.ts [hunk 1]
```diff
@@ -13,19 +13,61 @@ import {
...
+ * @see {@link Player}
+ */
 export const callPlayer = async <T>(
   fn: (player: Player) => Promise<T>
 ): Promise<T> => {
   return await PlayerWorld.shared.callPlayer(fn);
 };
 
+/**
+ * Calls the given function with the player instance and expects the given event to occur.
```
Comment (matamegger) [inline integration_test/playertesting/PlayerTesting.ts:56]: ? which given function

> Listens for the specified `Event` to be emitted and waits until the event is received

File: integration_test/playertesting/PlayerTesting.ts [hunk 2]
```diff
@@ -13,19 +13,61 @@ import {
...
+ * @param fn The function to call.
```
Comment (matamegger) [inline integration_test/playertesting/PlayerTesting.ts:57]:
```suggestion
```

File: integration_test/playertesting/PlayerTesting.ts [hunk 3]
```diff
@@ -36,6 +78,28 @@ export const expectEvent = async <T extends Event>(
   );
 };
 
+/**
+ * Calls the given function with the player instance and expects the given events to occur.
+ * @param fn The function to call.
```
Comment (matamegger) [inline integration_test/playertesting/PlayerTesting.ts:83]: Same as with the single event expectation

File: integration_test/playertesting/PlayerTesting.ts [hunk 4]
```diff
@@ -46,6 +110,22 @@ export const expectEvents = async (
   );
 };
 
+/**
+ * Calls the given function with the player instance and expects the given event to occur.
```
Comment (matamegger) [inline integration_test/playertesting/PlayerTesting.ts:114]: > Starts listening for the specified `SingleEventExpectationz before executing the passed `fn` function.
This is the race-condition-safe version of calling `callPlayer` and `expectEvent` after that. Useful when events are directly tied to calls in the `fn` function block and therefore synchronously emitted.

Maybe more in that direction.

File: integration_test/playertesting/PlayerTesting.ts (hunk 1 above)
Comment (rolandkakonyi) [reply integration_test/playertesting/PlayerTesting.ts]: oopsie, fixed in a4169682

File: integration_test/playertesting/PlayerTesting.ts (hunk 2 above)
Comment (rolandkakonyi) [reply integration_test/playertesting/PlayerTesting.ts]: fixed in a4169682

File: integration_test/playertesting/PlayerTesting.ts (hunk 3 above)
Comment (rolandkakonyi) [reply integration_test/playertesting/PlayerTesting.ts]: fixed in a4169682

File: integration_test/playertesting/PlayerTesting.ts (hunk 4 above)
Comment (rolandkakonyi) [reply integration_test/playertesting/PlayerTesting.ts]: fixed in a4169682

**Review by matamegger — APPROVED** [review]

**Pull request merged**